### PR TITLE
feat(upgrade): enable bulk upgrade for volumes and spc

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does / why we need it**:
This PR enables the bulk upgrade option for upgrade jobs where multiple volumes or spc can be upgraded in one job.

Updated job yaml:
```yaml
---
apiVersion: batch/v1
kind: Job
metadata:
  #VERIFY that you have provided a unique name for this upgrade job.
  #The name can be any valid K8s string for name. This example uses
  #the following convention: jiva-vol-<flattened-from-to-versions>
  name: jiva-vols-160190-busybox

  #VERIFY the value of namespace is same as the namespace where openebs components
  # are installed. You can verify using the command:
  # `kubectl get pods -n <openebs-namespace> -l openebs.io/component-name=maya-apiserver`
  # The above command should return status of the openebs-apiserver.
  namespace: openebs

spec:
  backoffLimit: 4
  template:
    spec:
      # VERIFY the value of serviceAccountName is pointing to service account
      # created within openebs namespace. Use the non-default account.
      # by running `kubectl get sa -n <openebs-namespace>`
      serviceAccountName: openebs-maya-operator
      containers:
      - name:  upgrade
        args:
        - "jiva-volume"

        # --from-version is the current version of the volume
        - "--from-version=1.6.0"

        # --to-version is the version desired upgrade version
        - "--to-version=1.9.0"

        #VERIFY that you have provided the correct list of jiva PV Name
        - "pvc-713e3bb6-afd2-11e9-8e79-42010a800065"
        - "pvc-02320288-22c8-44d4-b5a1-fc37acf2c597"
        
        #Following are optional parameters
        #Log Level
        - "--v=4"
        #DO NOT CHANGE BELOW PARAMETERS
        env:
        - name: OPENEBS_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        tty: true

        # the image version should be same as the --to-version mentioned above
        # in the args of the job
        image: quay.io/openebs/m-upgrade:1.9.0
        imagePullPolicy: Always
      restartPolicy: OnFailure
---
```

**Note: The old way of accepting pv name or spc name through the flags is still supported for upgrading one volume at a time.**
 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: cmd/upgrade/executor/cstor_spc.go [hunk 1]
```diff
@@ -50,11 +50,14 @@ func NewUpgradeCStorSPCJob() *cobra.Command {
 		Long:    cstorSPCUpgradeCmdHelpText,
 		Example: `upgrade cstor-spc --spc-name <spc-name>`,
```
Comment (vishnuitta) [inline cmd/upgrade/executor/cstor_spc.go:51]: this example needs a fix?

File: cmd/upgrade/executor/cstor_spc.go [hunk 2]
```diff
@@ -50,11 +50,14 @@ func NewUpgradeCStorSPCJob() *cobra.Command {
...
 		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.RunCStorSPCUpgradeChecks(cmd, args), util.Fatal)
 			options.resourceKind = "storagePoolClaim"
-			util.CheckErr(options.RunPreFlightChecks(cmd), util.Fatal)
-			util.CheckErr(options.RunCStorSPCUpgradeChecks(cmd), util.Fatal)
-			util.CheckErr(options.InitializeDefaults(cmd), util.Fatal)
-			util.CheckErr(options.RunCStorSPCUpgrade(cmd), util.Fatal)
+			if options.cstorSPC.spcName != "" {
+				singleCstorSPCUpgrade(cmd)
```
Comment (vishnuitta) [inline cmd/upgrade/executor/cstor_spc.go:56]: Cstor should be changed to CStor

File: cmd/upgrade/executor/cstor_spc.go [hunk 3]
```diff
@@ -50,11 +50,14 @@ func NewUpgradeCStorSPCJob() *cobra.Command {
 		Long:    cstorSPCUpgradeCmdHelpText,
 		Example: `upgrade cstor-spc --spc-name <spc-name>`,
 		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.RunCStorSPCUpgradeChecks(cmd, args), util.Fatal)
 			options.resourceKind = "storagePoolClaim"
-			util.CheckErr(options.RunPreFlightChecks(cmd), util.Fatal)
```
Comment (vishnuitta) [inline cmd/upgrade/executor/cstor_spc.go:54]: I don't need of passing 'cmd' to RunCStorSPCUpgradeChecks and RunCStorSPCUpgrade.. same may be the case with other checks.. can we remove that if not required

File: cmd/upgrade/executor/cstor_spc.go [hunk 4]
```diff
@@ -66,20 +69,38 @@ func NewUpgradeCStorSPCJob() *cobra.Command {
...
+		return errors.Errorf("Cannot execute upgrade job:" +
+			" neither spc-name flag is set nor spc name list is provided")
 	}
 
 	return nil
 }
 
 // RunCStorSPCUpgrade upgrades the given Jiva Volume.
 func (u *UpgradeOptions) RunCStorSPCUpgrade(cmd *cobra.Command) error {
+	klog.V(4).Infof("Started upgrading %s{%s} from %s to %s",
```
Comment (vishnuitta) [inline cmd/upgrade/executor/cstor_spc.go:97]: lets have similar message for completed upgrade also

File: cmd/upgrade/executor/cstor_spc.go (hunk 1 above)
Comment (shubham14bajpai) [reply cmd/upgrade/executor/cstor_spc.go]: Done

File: cmd/upgrade/executor/cstor_spc.go (hunk 2 above)
Comment (shubham14bajpai) [reply cmd/upgrade/executor/cstor_spc.go]: Done

File: cmd/upgrade/executor/cstor_spc.go (hunk 3 above)
Comment (shubham14bajpai) [reply cmd/upgrade/executor/cstor_spc.go:54]: Done

File: cmd/upgrade/executor/cstor_spc.go (hunk 4 above)
Comment (shubham14bajpai) [reply cmd/upgrade/executor/cstor_spc.go:100]: Done

**Pull request merged**